AssertionError: Failed to open youtube link

Before submitting a bug report, please be aware that your issue **must be reproducible** with all of the following, otherwise it is non-actionable, and we can not help you:
 - **Current repo**: run `git fetch && git status -uno` to check and `git pull` to update repo
 - **Common dataset**: coco.yaml or coco128.yaml
 - **Common environment**: Colab, Google Cloud, or Docker image. See https://github.com/ultralytics/yolov5#environments
 
If this is a custom dataset/training question you **must include** your `train*.jpg`, `test*.jpg` and `results.png` figures, or we can not help you. You can generate these with `utils.plot_results()`.


## 🐛 Bug
A clear and concise description of what the bug is.


## To Reproduce (REQUIRED)

Input:
python detect.py --source 'http://youtu.be/gmklhuYYL1k'

Output:
```
Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.25, device='', exist_ok=False, img_size=640, iou_thres=0.45, name='exp', project='runs/detect', save_conf=False, save_txt=False, source='https://youtu.be/gmklhuYYL1k', update=False, view_img=False, weights='yolov5s.pt')
YOLOv5 v4.0-16-g777032e torch 1.8.1+cu102 CUDA:0 (GeForce RTX 2080 Ti, 11019.4375MB)
                                          CUDA:1 (GeForce RTX 2080 Ti, 11019.4375MB)

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients, 17.0 GFLOPS
Traceback (most recent call last):
  File "detect.py", line 173, in <module>
    detect()
  File "detect.py", line 52, in detect
    dataset = LoadImages(source, img_size=imgsz)
  File "/home/ppark/yolo/yolov5/utils/datasets.py", line 132, in __init__
    raise Exception(f'ERROR: {p} does not exist')
Exception: ERROR: /home/ppark/yolo/yolov5/https:/youtu.be/gmklhuYYL1k does not exist
(yolov5) ppark@geosr:~/yolo/yolov5$ python detect.py --source 'http://youtu.be/gmklhuYYL1k'
Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.25, device='', exist_ok=False, img_size=640, iou_thres=0.45, name='exp', project='runs/detect', save_conf=False, save_txt=False, source='http://youtu.be/gmklhuYYL1k', update=False, view_img=False, weights='yolov5s.pt')
YOLOv5 v4.0-16-g777032e torch 1.8.1+cu102 CUDA:0 (GeForce RTX 2080 Ti, 11019.4375MB)
                                          CUDA:1 (GeForce RTX 2080 Ti, 11019.4375MB)

Fusing layers... 
Model Summary: 224 layers, 7266973 parameters, 0 gradients, 17.0 GFLOPS
1/1: http://youtu.be/gmklhuYYL1k... Traceback (most recent call last):
  File "detect.py", line 173, in <module>
    detect()
  File "detect.py", line 49, in detect
    dataset = LoadStreams(source, img_size=imgsz)
  File "/home/ppark/yolo/yolov5/utils/datasets.py", line 276, in __init__
    assert cap.isOpened(), f'Failed to open {s}'
AssertionError: Failed to open http://youtu.be/gmklhuYYL1k
```


## Expected behavior
A clear and concise description of what you expected to happen.


## Environment
If applicable, add screenshots to help explain your problem.

 - OS: [e.g. Ubuntu]
 - GPU [e.g. 2080 Ti]


## Additional context
Add any other context about the problem here.

--- a/utils/general.py
+++ b/utils/general.py
@@ -51,9 +51,18 @@ def get_latest_run(search_dir='.'):
     return max(last_list, key=os.path.getctime) if last_list else ''
 
 
-def isdocker():
+def is_docker():
     # Is environment a Docker container
     return Path('/workspace').exists()  # or Path('/.dockerenv').exists()
+
+
+def is_colab():
+    # Is environment a Google Colab instance
+    try:
+        import google.colab
+        return True
+    except Exception as e:
+        return False
 
 
 def emojis(str=''):
@@ -81,7 +90,7 @@ def check_git_status():
     print(colorstr('github: '), end='')
     try:
         assert Path('.git').exists(), 'skipping check (not a git repository)'
-        assert not isdocker(), 'skipping check (Docker image)'
+        assert not is_docker(), 'skipping check (Docker image)'
         assert check_online(), 'skipping check (offline)'
 
         cmd = 'git fetch && git config --get remote.origin.url'
@@ -138,7 +147,8 @@ def check_img_size(img_size, s=32):
 def check_imshow():
     # Check if environment supports image displays
     try:
-        assert not isdocker(), 'cv2.imshow() is disabled in Docker environments'
+        assert not is_docker(), 'cv2.imshow() is disabled in Docker environments'
+        assert not is_colab(), 'cv2.imshow() is disabled in Google Colab environments'
         cv2.imshow('test', np.zeros((1, 1, 3)))
         cv2.waitKey(1)
         cv2.destroyAllWindows()

